pkg/apis/core/validation/validation.go

```diff
@@ -2978,7 +2984,7 @@ func validateReadinessGates(readinessGates []core.PodReadinessGate, fldPath *fie
 	return allErrs
 }
 
-func validatePodDNSConfig(dnsConfig *core.PodDNSConfig, dnsPolicy *core.DNSPolicy, fldPath *field.Path) field.ErrorList {
+func validatePodDNSConfig(dnsConfig *core.PodDNSConfig, dnsPolicy *core.DNSPolicy, fldPath *field.Path, allowLiberalDNSConfig bool /* TODO: after 1.22, remove this */) field.ErrorList {
```

Comment:
plumb PodValidationOptions here instead of just the boolean

Reply:
Do we need to pass the whole PodValidationOptions?